crates/contract/src/call.rs

```diff
@@ -131,6 +131,23 @@ pub struct CallBuilder<T, P, D, N: Network = Ethereum> {
     transport: PhantomData<T>,
 }
 
+impl<T, P, D, N> CallBuilder<T, P, D, N>
+where
+    T: Transport + Clone,
+    P: Provider<T, N>,
+    D: CallDecoder + Send + Sync + Unpin,
+    N: Network,
+    Self: 'static,
+{
+    /// Retrieves a reference to the transaction request.
+    pub const fn request(&self) -> impl AsRef<N::TransactionRequest> + '_
+    where
+        <N as Network>::TransactionRequest: AsRef<<N as Network>::TransactionRequest>,
+    {
+        &self.request
```

Comment:
sorry, I mean that instead of `fn request() -> &N::TransactionRequest` you should write the function as an `impl AsRef<N::TransactionRequest>` block. that way it conforms to standard rust patterns

https://doc.rust-lang.org/std/convert/trait.AsRef.html

```
impl<T,P,D,N> AsRef<N::TransactionRequest> for CallBuilder<T, P, D, N>
where
    T: Transport + Clone,
    P: Provider<T, N>,
    D: CallDecoder + Send + Sync + Unpin,
    N: Network,
{
    ... 
}

Reply:
> sorry, I mean that instead of `fn request() -> &N::TransactionRequest` you should write the function as an `impl AsRef<N::TransactionRequest>` block. that way it conforms to standard rust patterns
> 
> https://doc.rust-lang.org/std/convert/trait.AsRef.html
> 
> ```
> impl<T,P,D,N> AsRef<N::TransactionRequest> for CallBuilder<T, P, D, N>
> where
>     T: Transport + Clone,
>     P: Provider<T, N>,
>     D: CallDecoder + Send + Sync + Unpin,
>     N: Network,
> {
>     ... 
> }
> ```

ty for your fast review! appreciate that lot